### PR TITLE
Misc mappings related to items and entities.

### PR DESCRIPTION
**Pull request by darkevilmac**
Mappings added: SmeltingRegistry, ItemFish.FishType, ItemFood saturation and hunger, Entity.addEffect, ItemStack metadata + getMetadata + getDamage

Please leave feedback, should getMaxDamage be getDurability, is hunger and saturation good enough for the ItemFood class or should it be hungerRestored and saturationRestored, etc.

### CODE REVIEW

File: mappings/net/minecraft/sortme/SmeltingRegistry.mapping [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	FIELD b inputOutputMap Ljava/util/Map;
+	FIELD c itemExperienceMap Ljava/util/Map;
+	METHOD a getRegistry ()Lnone/agq;
+	METHOD a registerSmeltingRecipe (Lnone/aeo;Lnone/aeq;F)V
+		ARG 0 input
+		ARG 1 output
+		ARG 2 exp
+	METHOD a getSmeltingResult (Lnone/aeq;)Lnone/aeq;
+		ARG 0 input
+	METHOD a canDoSmelting (Lnone/aeq;Lnone/aeq;)Z
```
Comment (asiekierka) [inline mappings/net/minecraft/sortme/SmeltingRegistry.mapping]: Not sure here... canSmeltWith _might_ be better.

File: mappings/net/minecraft/sortme/SmeltingRegistry.mapping [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	METHOD a registerSmeltingRecipe (Lnone/aeq;Lnone/aeq;F)V
+		ARG 0 input
+		ARG 1 output
+		ARG 2 exp
+	METHOD a registerSmeltingRecipe (Lnone/akw;Lnone/aeq;F)V
+		ARG 0 input
+		ARG 1 output
+		ARG 2 exp
+	METHOD b getInputOutputMap ()Ljava/util/Map;
+	METHOD b getItemExperience (Lnone/aeq;)F
```
Comment (asiekierka) [inline mappings/net/minecraft/sortme/SmeltingRegistry.mapping]: Maybe getExperienceForSmelting? getExperience?

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 3]
```diff
@@ -49,6 +50,8 @@ CLASS none/aeq net/minecraft/item/ItemStack
 		ARG 0 amount
 	METHOD g subtractStackSize (I)V
 		ARG 0 amount
+	METHOD i getMetadata ()I
+	METHOD j getMetadata2 ()I
```
Comment (asiekierka) [inline mappings/net/minecraft/item/ItemStack.mapping]: Nonononono. I'm pretty sure they're not both metadata. Could you look at the calls?

File: mappings/net/minecraft/sortme/SmeltingRegistry.mapping (hunk 2 above)
Comment (darkevilmac) [inline mappings/net/minecraft/sortme/SmeltingRegistry.mapping]: does getSmeltingExperience sound good?

File: mappings/net/minecraft/item/Item.mapping [hunk 4]
```diff
@@ -83,6 +84,7 @@ CLASS none/aeo net/minecraft/item/Item
 		ARG 0 maxStackSize
 	METHOD d getItemById (Ljava/lang/String;)Lnone/aeo;
 		ARG 0 id
+	METHOD e setMaxDamage (I)Lnone/aeo;
```
Comment (asiekierka) [inline mappings/net/minecraft/item/Item.mapping]: Comment request: maxDamage or durability?

File: mappings/net/minecraft/item/ItemFood.mapping [hunk 5]
```diff
@@ -1 +1,9 @@
 CLASS none/aek net/minecraft/item/ItemFood
+	FIELD b hunger I
```
Comment (asiekierka) [inline mappings/net/minecraft/item/ItemFood.mapping]: hunger or hungerRestored? Same with saturation

File: mappings/net/minecraft/item/impl/ItemFish.mapping [hunk 6]
```diff
@@ -1,5 +1,35 @@
...
+		METHOD a getTypeFromId (I)Lnone/aeh$a;
+			ARG 0 id
+		METHOD a getTypeFromStack (Lnone/aeq;)Lnone/aeh$a;
+			ARG 0 itemStack
+		METHOD b getName ()Ljava/lang/String;
+		METHOD c getRawHunger ()I
+		METHOD d getRawSaturation ()F
+		METHOD e getCookedHunger ()I
+		METHOD f getCookedSaturation ()F
+		METHOD g canCook ()Z
```
Comment (asiekierka) [inline mappings/net/minecraft/item/impl/ItemFish.mapping]: Comment request: canCook or isCookable?

Comment (copygirl) [inline mappings/net/minecraft/item/impl/ItemFish.mapping]: I'd prefer `isCookable`, and in general use `isXYZ` for every similar simple method returning a bool (`cookable` would be used for fields?). `canXYZ` should be used for methods that take more parameters, like for example `canEat(ItemStack, EntityPlayer)`.

File: mappings/net/minecraft/item/Item.mapping (hunk 4 above)
Comment (marysaka) [inline mappings/net/minecraft/item/Item.mapping]: I personally prefer durability because it's more clear.

Comment (copygirl) [inline mappings/net/minecraft/item/Item.mapping]: I'd personally like durability the most, though an item's "damage" is used for more than just that, right? (I suppose :+1: this comment if you want `durability` and :-1: if `maxDamage`.)

File: mappings/net/minecraft/item/ItemFood.mapping (hunk 5 above)
Comment (asiekierka) [inline mappings/net/minecraft/item/ItemFood.mapping]: @copygirl Saying "yes" to an "A or B" question is not helping me at all!

Comment (copygirl) [inline mappings/net/minecraft/item/ItemFood.mapping]: It's currently hunger, yours suggests an alternative, so :+1: could only possibly mean I support your alternative :stuck_out_tongue:

File: mappings/net/minecraft/item/Item.mapping [hunk 7]
```diff
@@ -38,7 +39,7 @@ CLASS none/aeo net/minecraft/item/Item
 		ARG 0 tab
 	METHOD a getRawIdByItem (Lnone/aeo;)I
 		ARG 0 item
-	METHOD a (Lnone/aeo;Lnone/adq;Lnone/dc;)V
+	METHOD a addSubtypesToCreativeTab (Lnone/aeo;Lnone/adq;Lnone/dc;)V
```
Comment (asiekierka) [inline mappings/net/minecraft/item/Item.mapping]: In Block we use "METHOD a addStacksForDisplay (Lnone/aeo;Lnone/adq;Lnone/dc;)V" and I think it makes more sense personally.

File: mappings/net/minecraft/sortme/SmeltingRegistry.mapping [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	FIELD b inputOutputMap Ljava/util/Map;
+	FIELD c itemExperienceMap Ljava/util/Map;
+	METHOD a getRegistry ()Lnone/agq;
+	METHOD a registerSmeltingRecipe (Lnone/aeo;Lnone/aeq;F)V
+		ARG 0 input
+		ARG 1 output
+		ARG 2 exp
+	METHOD a getSmeltingResult (Lnone/aeq;)Lnone/aeq;
+		ARG 0 input
+	METHOD a canSmeltWith (Lnone/aeq;Lnone/aeq;)Z
```
Comment (asiekierka) [inline mappings/net/minecraft/sortme/SmeltingRegistry.mapping]: On second thought, canSmeltWith is not right either. Can you help me out?

**Pull request merged**